src/software/thunderscope/robot_communication.py

```diff
@@ -104,34 +137,181 @@ def __init__(
             except Exception:
                 raise Exception(f"Invalid Estop found at location {self.estop_path}")
 
-    def setup_for_fullsystem(self) -> None:
+        self.print_current_network_config()
+
+    def setup_for_fullsystem(
+        self,
+        referee_interface: Optional[str] = None,
+        vision_interface: Optional[str] = None,
+        force_reconnect: bool = False,
+    ) -> None:
         """
-        Sets up a listener for SSL vision and referee data, and connects all robots to fullsystem as default
+        Sets up a listener for SSL vision and referee data
+
+        :param referee_interface: the interface to listen for referee data
+        :param vision_interface: the interface to listen for vision data
+        :param force_reconnect: whether to force a reconnection regardless of whether our stored state says we are
+        already connected
         """
-        self.receive_ssl_wrapper = tbots_cpp.SSLWrapperPacketProtoListener(
-            SSL_VISION_ADDRESS,
-            SSL_VISION_PORT,
-            lambda data: self.__forward_to_proto_unix_io(SSL_WrapperPacket, data),
-            True,
+        new_referee_interface = (
+            referee_interface
+            if referee_interface
+            else self.current_network_config.referee_interface
+        )
+        new_vision_interface = (
+            vision_interface
+            if vision_interface
+            else self.current_network_config.vision_interface
         )
 
-        self.receive_ssl_referee_proto = tbots_cpp.SSLRefereeProtoListener(
-            SSL_REFEREE_ADDRESS,
-            self.referee_port,
-            lambda data: self.current_proto_unix_io.send_proto(Referee, data),
-            True,
+        change_referee_interface = (
+            new_referee_interface != self.current_network_config.referee_interface
+        ) and (new_referee_interface != DISCONNECTED)
+        change_vision_interface = (
+            new_vision_interface != self.current_network_config.vision_interface
+        ) and (new_vision_interface != DISCONNECTED)
+
+        if force_reconnect:
+            change_referee_interface = (
+                True
+                if new_referee_interface != DISCONNECTED
+                else change_referee_interface
+            )
+            change_vision_interface = (
+                True
+                if new_vision_interface != DISCONNECTED
+                else change_vision_interface
+            )
```

Comment:
why is this logic so weird?

it's because `setup_for_fullsystem` gets called in `thunderscope_main` (`RobotCommuncation` doesn't know if it needs to connect to referee or vision messages in the constructor). The constructor stores the `--interface` args for its `referee` and `vision` interface cache without connecting to those interfaces but later needs to connect to it once it gets a signal from `thunderscope_main`. `force_reconnect` forces us to bind to the interface even if the cache says we're already connected

Reply:
nvm I simplified the logic. there was a better way